File: dawaswiftdriver/src/main/java/com/dev/dawaswiftdriver/RequestCardFragment.java
package com.dev.dawaswiftdriver;

import android.content.Context;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class RequestCardFragment extends Fragment {
    Request request;
    private CountDownTimer countDownTimer;
    private OnFragmentInteractionListener mListener;
    private static final String ARG_REQUEST = "request";

    public RequestCardFragment() {

    }

    public static RequestCardFragment newInstance(Request request) {
        RequestCardFragment fragment = new RequestCardFragment();
        Bundle args = new Bundle();
        args.putSerializable(ARG_REQUEST, request);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            request = (Request) getArguments().getSerializable(ARG_REQUEST);
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
       return inflater.inflate(R.layout.fragment_request_card, container, false);

//        binding = DataBindingUtil.inflate(inflater,R.layout.fragment_request_card, container,false);
//        binding.setRequest(request);
//        countDownTimer = new CountDownTimer(5 * 60000, 50) {
//            @Override
//            public void onTick(long millisUntilFinished) {
//            }
//
//            @Override
//            public void onFinish() {
//                if(mListener != null && request != null)
//                    mListener.onDecline(request);
//            }
//        };
//        countDownTimer.start();
//        if(getResources().getBoolean(R.bool.use_miles))
//            binding.textDistance.setText(getString(R.string.unit_distance_miles,request.distance / 1609.344f));
//        else
//            binding.textDistance.setText(getString(R.string.unit_distance,request.distance / 1000f));
//        if(getResources().getBoolean(R.bool.use_miles))
//            binding.textFromYou.setText(getString(R.string.unit_distance_miles,request.fromDriver / 1000f));
//        else
//            try {
//                binding.textFromYou.setText(getString(R.string.unit_distance, request.fromDriver / 1000f));
//                binding.buttonAccept.setOnClickListener(view -> {
//                    countDownTimer.cancel();
//                    mListener.onAccept(request);
//                });
//                binding.buttonDecline.setOnClickListener(view -> {
//                    countDownTimer.cancel();
//                    mListener.onDecline(request);
//                });
//            }catch (Exception nm){
//                nm.printStackTrace();
//            }
//        return binding.getRoot();
//





    }

    @Override
    public void onViewCreated(@NonNull View vieww, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(vieww, savedInstanceState);

        countDownTimer = new CountDownTimer(5 * 60000, 50) {
                        @Override
            public void onTick(long millisUntilFinished) {
            }

            @Override
            public void onFinish() {
                if(mListener != null && request != null)
                    mListener.onDecline(request);
            }
        };
        countDownTimer.start();
        TextView distamce=vieww.findViewById(R.id.text_distance);
        TextView from=vieww.findViewById(R.id.text_from);
        TextView to=vieww.findViewById(R.id.text_to);
        TextView cost=vieww.findViewById(R.id.text_cost);
        TextView fromyou=vieww.findViewById(R.id.text_from_you);
        Button buttonAccept=vieww.findViewById(R.id.button_accept);
        Button buttonDecline=vieww.findViewById(R.id.button_decline);


        from.setText(request.travel.getPickupAddress());
        to.setText(request.travel.getDestinationAddress());
        cost.setText(request.cost.toString());


        distamce.setText(getString(R.string.unit_distance,request.distance / 1000f));
        fromyou.setText(getString(R.string.unit_distance, request.fromDriver / 1000f));
        buttonAccept.setOnClickListener(view -> {
                    countDownTimer.cancel();
                    mListener.onAccept(request);
                });
        buttonDecline.setOnClickListener(view -> {
            countDownTimer.cancel();
            mListener.onDecline(request);
        });

    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnFragmentInteractionListener) {
            mListener = (OnFragmentInteractionListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    public interface OnFragmentInteractionListener {
        void onAccept(Request request);
        void onDecline(Request request);
    }
}

File: dawaswiftdriver/src/main/java/com/dev/dawaswiftdriver/views/requests/RequestCardFragment.kt
package com.dev.dawaswiftdriver.views.requests

import android.content.Context
import android.os.Bundle
import android.os.CountDownTimer
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.dev.common.models.driver.requests.TripRequest
import com.dev.dawaswiftdriver.R

class RequestCardFragment : Fragment() {
    internal var request: TripRequest? = null
    private var countDownTimer: CountDownTimer? = null
    private var mListener: OnFragmentInteractionListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            request = arguments!!.getSerializable(ARG_REQUEST) as TripRequest
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_request_card, container, false)


    }

    override fun onViewCreated(vieww: View, savedInstanceState: Bundle?) {
        super.onViewCreated(vieww, savedInstanceState)

        countDownTimer = object : CountDownTimer((5 * 60000).toLong(), 50) {
            override fun onTick(millisUntilFinished: Long) {}

            override fun onFinish() {
//                if (mListener != null && request != null)
//                   mListener!!.onDecline(request!!)
            }
        }
        countDownTimer!!.start()
        val distamce = vieww.findViewById<TextView>(R.id.text_distance)
        val from = vieww.findViewById<TextView>(R.id.pick_up)
        val to = vieww.findViewById<TextView>(R.id.drop_off)
        val cost = vieww.findViewById<TextView>(R.id.text_earning)
        val buttonAccept = vieww.findViewById<Button>(R.id.button_accept)
        val buttonDecline = vieww.findViewById<Button>(R.id.button_decline)

        var listOfLocations: MutableList<String> = ArrayList()
        for (pickup in request?.pickUpPoints!!) {

            listOfLocations.add(pickup.location!!)
        }

        from.text = listOfLocations.joinToString()

        to.text = request?.dropOffPoint?.location
        cost.text = request?.earnings?.toString()


        distamce.text = getString(R.string.unit_distance, request!!.distance!! / 1000f)
        buttonAccept.setOnClickListener { view ->
            countDownTimer!!.cancel()
            mListener!!.onAccept(request as TripRequest)
        }
        buttonDecline.setOnClickListener { view ->
            countDownTimer!!.cancel()
            mListener!!.onDecline(request as TripRequest)
        }

    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is OnFragmentInteractionListener) {
            mListener = context
        } else {
            throw RuntimeException("$context must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    interface OnFragmentInteractionListener {
        fun onAccept(request: TripRequest)
        fun onDecline(request: TripRequest)
    }

    companion object {
        private val ARG_REQUEST = "request"

        fun newInstance(request: TripRequest): RequestCardFragment {
            val fragment = RequestCardFragment()
            val args = Bundle()
            args.putSerializable(ARG_REQUEST, request)
            fragment.arguments = args
            return fragment
        }
    }
}
